tests/vmi_multus_test.go

```diff
@@ -1008,16 +1046,74 @@ var _ = Describe("[Serial]Macvtap", func() {
 
 		BeforeEach(func() {
 			macAddress := "02:03:04:05:06:07"
-			macvtapVMI = createCirrosVMIWithMacvtapDefinedMAC(virtClient, macvtapNetworkName, macAddress)
+			clientVMI = createCirrosVMIRandomNode(virtClient, macvtapNetworkName, macAddress)
 		})
 
 		It("should be successful when the VMI MAC address is defined in its spec", func() {
 			By("starting the migration")
-			migration := tests.NewRandomMigration(macvtapVMI.GetName(), macvtapVMI.GetNamespace())
+			migration := tests.NewRandomMigration(clientVMI.GetName(), clientVMI.GetNamespace())
 			migrationUID := tests.RunMigrationAndExpectCompletion(virtClient, migration, migrationWaitTime)
 
 			// check VMI, confirm migration state
-			tests.ConfirmVMIPostMigration(virtClient, macvtapVMI, migrationUID)
+			tests.ConfirmVMIPostMigration(virtClient, clientVMI, migrationUID)
+		})
+
+		Context("with live traffic", func() {
+			var serverVMI *v1.VirtualMachineInstance
+			var serverIP string
+
+			getVMMacvtapIfaceIP := func(vmi *v1.VirtualMachineInstance, macAddress string) (string, error) {
+				var vmiIP string
+				err := wait.PollImmediate(time.Second, 2*time.Minute, func() (done bool, err error) {
+					vmi, err := virtClient.VirtualMachineInstance(vmi.GetNamespace()).Get(vmi.GetName(), &v13.GetOptions{})
+					if err != nil {
+						return false, err
+					}
+
+					for _, iface := range vmi.Status.Interfaces {
+						if iface.MAC == macAddress {
+							vmiIP = iface.IP
+							return true, nil
+						}
+					}
+
+					return false, nil
+				})
+				if err != nil {
+					return "", err
+				}
+
+				// code below should be removed if / when KubeVirt honors its
+				// API and returns an IP on the Interface.IP attribute
+				// Track the following PR to know when to update:
+				//   - https://github.com/kubevirt/kubevirt/pull/4217
+				ipAndNetworkLength := strings.Split(vmiIP, "/")
+				if len(ipAndNetworkLength) != 2 {
+					return "", fmt.Errorf("could not compute the IP address from the CIDR")
+				}
+				return ipAndNetworkLength[0], nil
+			}
+
+			BeforeEach(func() {
+				macAddress := "02:03:04:05:06:aa"
+
+				serverVMI = createFedoraVMIRandomNode(virtClient, macvtapNetworkName, macAddress)
+				Expect(serverVMI.Status.Interfaces).NotTo(BeEmpty(), "a migrate-able VMI must have network interfaces")
+
+				serverIP, err = getVMMacvtapIfaceIP(serverVMI, macAddress)
+				Expect(err).NotTo(HaveOccurred(), "should have managed to figure out the IP of the server VMI")
+			})
+
+			BeforeEach(func() {
+				Expect(libnet.PingFromVMConsole(clientVMI, serverIP)).To(Succeed(), "connectivity is expected *before* migrating the VMI")
+			})
+
+			It("should keep connectivity after a migration", func() {
+				migration := tests.NewRandomMigration(serverVMI.GetName(), serverVMI.GetNamespace())
+				_ = tests.RunMigrationAndExpectCompletion(virtClient, migration, migrationWaitTime)
```

Comment:
Why is `_` needed?

Reply:
It is not needed, but I'm communicating to the reader that the `tests.RunMigrationAndExpectCompletion` retuns something that I do *not* care about, which adds more info that omitting it.

Reply:
I find this explicit usage confusing, as if you are trying to tell something that is not obvious and the reader should care.
Well, if the returned value is not important, why do I need to think about it? Is this a recommended Go pattern?

Reply:
I'm just pointing out that it is *not* obvious at all that  `tests.RunMigrationAndExpectCompletion(virtClient, migration, migrationWaitTime)` returns any value. It could return an important value, that the user is unaware of, but since it got copied from somewhere, it is not being handled. 

If I do `_ = tests.RunMigrationAndExpectCompletion(virtClient, migration, migrationWaitTime)` on the other hand, I'm explicitly indicating that there is a return value, and that in this scope it is not needed, and we can safely ignore it.

To me, one of those forms is clearly preferable. What is it that you find confusing ? I am having a hard time sharing that feeling.

Reply:
Your reasoning would have required this pattern to be a language requirement or a project best practice (e.g. when you close a file, to use `_` as well).

It is confusing because it is not needed by the language, it is much clearer that when you do not use the return value, you do not care about it. The usage of `_` has a specific purpose in cases where you must accept the value, otherwise the compilation fails.
So the confusion is the unorthodox usage of this here.

Reply:
Clearer in the sense that the reader is mislead into thinking a function does not return anything vs knowing something is returned & ignored ?